runtime/v2/runner/src/main/java/com/walmartlabs/concord/runtime/v2/runner/vm/SaveLastErrorCommand.java

```diff
@@ -54,14 +56,24 @@ public void eval(Runtime runtime, State state, ThreadId threadId) throws Excepti
         Map<String, Object> currentOut = persistenceService.loadPersistedFile(Constants.Files.OUT_VALUES_FILE_NAME, in -> om.readValue(in, MAP_TYPE));
 
         Map<String, Object> outValues = new HashMap<>(currentOut != null ? currentOut : Collections.emptyMap());
-        outValues.put(Constants.Context.LAST_ERROR_KEY, createMapper().convertValue(e, Map.class));
-
+        outValues.put(Constants.Context.LAST_ERROR_KEY, serialize(e));
         persistenceService.persistFile(Constants.Files.OUT_VALUES_FILE_NAME,
                 out -> om.writeValue(out, outValues));
 
         throw e;
     }
 
+    private static Map<String, Object> serialize(Exception e) {
+        ObjectMapper mapper = createMapper();
+
+        try {
+            return mapper.convertValue(e, MAP_TYPE);
+        } catch (Exception ex) {
+            // ignore ex
+            return mapper.convertValue(Collections.singletonMap("message", e.getMessage()), MAP_TYPE);
```

Comment:
Do we really need `convertValue` here? Can we just do `return Collections.singletonMap("message", e.getMessage())`?

Reply:
Ah, it's to apply this `IntSequenceGenerator` business, right?

Reply:
oops. yep, we can just return `singletonMap`